docs/CIP/CIP-XXXX/README.md

```diff
@@ -0,0 +1,272 @@
+
+Since our wallet-optimized protocol defines that server-sent messages include the `point` message partial, the client can check if its last `point` is greater than any subsequent message `point` from the server, without the need to introduce a `rollback` specific message partial. This assumes that the client keeps a local copy of the mutuable suffix of the blockchain defined by the security parameter.
+
+#### Connection Management
+
+Clients are expected to reconnect on error, when providing invalid `point`s for which a server cannot find an intersection.
+Clients may send multiple [`subscribe`](./messages/client/subscribe.md) messages for different accounts, but subsequently take responsibility to coordinate server-sent events by time to their possibly differently synchronized accounts. When receiving a message that references a `point` that the first account has already seen, the client only applies it for its second account for example.
+
+### Versioning
+
+In our protocol, the client sends its list of supported versions as part of the initial [`subscribe`](./messages/client/subscribe.md) message. Upon receiving this information, the server selects the highest mutually supported version and associates it with the client.
+This method requires the server to implement version-based message construction, allowing it to tailor its responses to each client's supported version.
+
+### Encoding
+
+Our proposed communication protocol **adopts blockchain-native encoding standards** to ensure compatibility and efficiency. This approach leverages existing conventions familiar to clients while minimizing conversion overhead on the server-side.
+
+#### Transaction Hash Encoding
+
+Transaction hashes are typically encoded using **hexadecimal representation**.
+
+#### Address Encoding
+
+Address encoding varies depending on the blockchain network. In Cardano the protocol servers bech32 encoded addresses.
+
+### Extensions
+
+Clients can turn on certain `extensions` by setting a respective flags in their subscription [`config](./messages/client/subscribe.md#subscribe-example) which tells the server to for example:
+
+- to enrich transaction output asset values by their respective off-chain metadata such as decimal offsets
+- to resolve transaction inputs (transaction output references) as transaction outputs
+
+## Rationale: how does this CIP achieve its goals?
+
+A primary consideration for the protocol design was to minimize message round trips between client and server as well as the amount of work to be done by the server for each client. Thus, the API's authentication does not require an initial message exchange but rather defines a standard verifable method for the server to either drop a client early and proceed to publish its relevant data.
+
+### Client Typed Messages
+
+- [`heartbeat`](./messages/client/heartbeat.md): Connection management/ keep alive
```

Comment:
This is a transport-specific message. Are we sure it's even needed with WS? I'd expect the connection to be maintained by either the protocol or lower level libraries.
```suggestion
```

Reply:
WebSocket connections can drop due to network issues or infrastructure timeouts. Proxies and firewalls may terminate idle WebSocket connections too.

Reply:
Infrastructure can be configured. Anyway, even if this is actually required for WS, it's transport-specific. So based on https://github.com/will-break-it/wallet-architecture/pull/5#discussion_r1678761352 this is probably not the right place to define this message? :thinking:

Reply:
You could make note that since the API supports extensions,  transport-specific messages can be implemented.